layer2/CifFile.cpp

```diff
@@ -268,12 +268,20 @@ bool cif_file::parse(char*&& p) {
       if (*p)
         *(p++) = 0;
       prev = *p;
-    } else if (*p == ';' && islinefeed(prev)) { // will NULL the line feed before the closing semicolon
+    } else if (*p == ';' && islinefeed(prev)) {
+      // multi-line tokens start with ";" and end with "\n;"
+      // multi-line tokens cannot be keys, only values.
       keypossible.push_back(false);
       tokens.push_back(p + 1);
+      // advance until `\n;`
       while (*++p && !(islinefeed(*p) && p[1] == ';'));
+      // step to next line and null the line feed
       if (*p) {
         *p = 0;
+        // \r\n on Windows)
+        if (p - 1 > tokens.back() && islinefeed(*(p - 1))) {
```

Comment:
I think this should be `*(p - 1) == '\r'` instead of `islinefeed(*(p - 1))`, otherwise this would also trim an empty line with unix line ending.
```
;next line is intentionally blank

;
```

Reply:
Good point!